frame/contracts/src/benchmarking/mod.rs

```diff
@@ -1866,20 +1866,14 @@ benchmarks! {
 	// It generates different private keys and signatures for the message "Hello world".
 	seal_ecdsa_recover {
 		let r in 0 .. API_BENCHMARK_BATCHES;
-		use rand::SeedableRng;
-		let mut rng = rand_pcg::Pcg32::seed_from_u64(123456);
 
 		let message_hash = sp_io::hashing::blake2_256("Hello world".as_bytes());
+		let key_type = sp_core::crypto::KeyTypeId(*b"code");
+		let pub_key = sp_io::crypto::ecdsa_generate(key_type, None);
```

Comment:
This changes the behavior so that all signatures are generated from the same key. Please move it back into the `map`.

Reply:
I just moved it out because the signature contains an aleatory component, i.e. all the signatures are different.
I also thought that the benchmark objective is to measure the recovery the PK from the signature, so this detail IMO is actually not important.
Nevertheless I can move it back to the map

Reply:
Yeah the benchmark should measure the recovery. However, I thought it would be best to use different keys just to be sure to not trigger any optimization pass there.